src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -99,7 +100,8 @@ def register_options(cls, register):
                 "independently."
                 "\n\nUsually this option should be enabled because it can result in far fewer "
                 "resolves."
-                "\n\nRequires [python].requirement_constraints to be set."
+                "\n\nRequires `[python].requirement_constraints` to be set. If you're using the "
+                "improved `[python].enable_resolves` feature, this is always the behavior."
```

Comment:
I think if we do deprecate this, we need to first deprecate `--requirement-constraints`. At that point, we keep this option to avoid churn, but it will no-op once `--requirement-constraints` is removed and then we can deprecate it.